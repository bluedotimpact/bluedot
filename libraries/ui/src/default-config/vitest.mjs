/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
