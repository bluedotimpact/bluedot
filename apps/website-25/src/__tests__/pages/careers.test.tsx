import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import CareersPage from '../../pages/careers';

describe('CareersPage', () => {
  test('should render correctly', () => {
    const { container } = render(<CareersPage />);
    expect(container).toMatchSnapshot();
  });
});
