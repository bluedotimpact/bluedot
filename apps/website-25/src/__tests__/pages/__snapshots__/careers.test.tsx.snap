// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CareersPage > should render the error message correctly 1`] = `
<div>
  <div>
    <div
      class="hero-section bg-bluedot-darker flex flex-row justify-center items-center w-full px-spacing-x"
    >
      <div
        class="hero-section__content max-w-[865px] mt-[82px] py-28"
      >
        <h1
          class="hero-section__title text-on-dark text-center"
        >
          Join us in our mission to ensure humanity safely navigates the transition to transformative AI.
        </h1>
      </div>
    </div>
    <section
      class="section section-body intro-section"
    >
      <div
        class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
      >
        <div
          class="section-heading__content flex-1 flex flex-col gap-2"
        >
          <h2
            class="section-heading__title relative"
          >
            Our culture
          </h2>
        </div>
      </div>
      <div
        class="section__body"
      >
        <div
          class="intro-section__container flex lg:flex-row flex-col gap-space-between"
        >
          <div
            class="intro-section__text-container flex flex-col gap-5"
          >
            <p>
              Humanity faces unprecedented challenges from transformative AI (TAI) which could profoundly reshape our future. Current AI systems already exhibit concerning behaviours, and as they become more powerful, misalignment between their actions, human intentions and societal values could lead to catastrophic outcomes.
            </p>
            <p>
              BlueDot Impact exists to safeguard humanity from these risks by building and accelerating the AI safety field.  We’ve trained over 4,000 professionals and helped hundreds land roles at organisations like Anthropic and the UK’s AI Safety Institute. We’ve built a strong foundation of deep knowledge in how to communicate complex concepts, close relationships across the field, and expertise in talent development. However, urgency and the magnitude of AI risk demands we do much more.
            </p>
          </div>
          <img
            alt="BlueDot Impact team"
            class="intro-section__image max-w-[570px] w-full h-auto rounded-2xl"
            src="/images/team/team_1.jpg"
          />
        </div>
      </div>
    </section>
    <div
      class="slide-list w-full flex flex-col values-section section-body"
    >
      <div
        class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
      >
        <div
          class="section-heading__content flex-1 flex flex-col gap-2"
        >
          <h2
            class="section-heading__title relative"
          >
            Our values
          </h2>
        </div>
        <div
          class="slide-list__nav--header items-center gap-2 ml-auto mt-auto flex"
        >
          <button
            aria-label="Previous slide"
            class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-none focus:ring-2 focus:ring-bluedot-light"
            disabled=""
            type="button"
          >
            <svg
              class="slide-list__nav-icon size-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <button
            aria-label="Next slide"
            class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-none focus:ring-2 focus:ring-bluedot-light"
            type="button"
          >
            <svg
              class="slide-list__nav-icon size-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 5l7 7-7 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="slide-list__content flex flex-col lg:flex-row gap-space-between"
      >
        <div
          class="slide-list__container relative overflow-hidden flex-1 flex flex-col gap-space-between"
        >
          <div
            class="slide-list__nav--container items-center justify-end gap-2 hidden"
          >
            <button
              aria-label="Previous slide"
              class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-none focus:ring-2 focus:ring-bluedot-light"
              disabled=""
              type="button"
            >
              <svg
                class="slide-list__nav-icon size-5"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 19l-7-7 7-7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <button
              aria-label="Next slide"
              class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-none focus:ring-2 focus:ring-bluedot-light"
              type="button"
            >
              <svg
                class="slide-list__nav-icon size-5"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 5l7 7-7 7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
          <div
            class="slide-list__gradient-overlay absolute inset-0 pointer-events-none z-10 flex"
          >
            <div
              class="slide-list__gradient-right absolute inset-y-0 right-0 w-8 bg-gradient-to-l from-cream-normal to-transparent opacity-50"
            />
          </div>
          <div
            class="slide-list__slides flex overflow-x-scroll transition-transform duration-300 gap-space-between"
          >
            <div
              class="slide-list__slide flex-shrink-0"
              style=""
            >
              <div
                class="card flex flex-col items-start transition-transform duration-200 values-section__value"
              >
                <div
                  class="card__image-container w-full mb-4"
                >
                  <img
                    alt="Think hard, act fast, fail faster"
                    class="card__image w-full max-h-full object-cover rounded-radius-md "
                    src="/images/values/think.png"
                  />
                </div>
                <div
                  class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
                >
                  <div
                    class="card__text"
                  >
                    <p
                      class="card__title subtitle-sm mb-2"
                    >
                      Think hard, act fast, fail faster
                    </p>
                    <p
                      class="card__subtitle "
                    >
                      We think critically about our goals and the best path to achieve them. We learn by building rapid experiments, failing fast, measuring the results, and updating.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="slide-list__slide flex-shrink-0"
              style=""
            >
              <div
                class="card flex flex-col items-start transition-transform duration-200 values-section__value"
              >
                <div
                  class="card__image-container w-full mb-4"
                >
                  <img
                    alt="Care personally, challenge directly"
                    class="card__image w-full max-h-full object-cover rounded-radius-md "
                    src="/images/values/care.png"
                  />
                </div>
                <div
                  class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
                >
                  <div
                    class="card__text"
                  >
                    <p
                      class="card__title subtitle-sm mb-2"
                    >
                      Care personally, challenge directly
                    </p>
                    <p
                      class="card__subtitle "
                    >
                      We care about our team and our community, and we hold everyone to high standards.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="slide-list__slide flex-shrink-0"
              style=""
            >
              <div
                class="card flex flex-col items-start transition-transform duration-200 values-section__value"
              >
                <div
                  class="card__image-container w-full mb-4"
                >
                  <img
                    alt="Obsessed with empowering problem-solvers"
                    class="card__image w-full max-h-full object-cover rounded-radius-md "
                    src="/images/values/powering.png"
                  />
                </div>
                <div
                  class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
                >
                  <div
                    class="card__text"
                  >
                    <p
                      class="card__title subtitle-sm mb-2"
                    >
                      Obsessed with empowering problem-solvers
                    </p>
                    <p
                      class="card__subtitle "
                    >
                      We exist to support others to solve the world's biggest problems. We go above and beyond to accelerate our community's impact.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <section
      class="section section-body careers-section"
    >
      <div
        class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
      >
        <div
          class="section-heading__content flex-1 flex flex-col gap-2"
        >
          <h2
            class="section-heading__title relative"
          >
            Careers at BlueDot Impact
          </h2>
        </div>
      </div>
      <div
        class="section__body"
      >
        <div
          class="careers-section__container flex flex-col gap-8"
        >
          <div
            class="careers-section__listing"
          >
            <div
              class="careers-section__card--desktop w-full flex flex-row items-center justify-between p-8 container-lined"
            >
              <strong
                class="careers-section__title basis-[33%]"
              >
                AI Alignment Project Judge
              </strong>
              <p
                class="careers-section__location"
              >
                London, Remote
              </p>
              <p
                class="careers-section__type"
              >
                Permanent
              </p>
              <a
                class="cta-button flex items-center justify-center rounded-radius-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter careers-section__cta-button"
                data-testid="cta-link"
                href="https://bluedot.org/ai-alignment-project-judge/"
              >
                <span
                  class="cta-button__text"
                >
                  Apply now
                </span>
                <span
                  class="cta-button__chevron ml-3"
                >
                  <img
                    alt="→"
                    class="cta-button__chevron-icon size-2"
                    src="/icons/chevron_blue.svg"
                  />
                </span>
              </a>
            </div>
          </div>
          <div
            class="careers-section__listing"
          >
            <div
              class="careers-section__card--desktop w-full flex flex-row items-center justify-between p-8 container-lined"
            >
              <strong
                class="careers-section__title basis-[33%]"
              >
                Course Operations Specialist
              </strong>
              <p
                class="careers-section__location"
              >
                London, Remote
              </p>
              <p
                class="careers-section__type"
              >
                Permanent
              </p>
              <a
                class="cta-button flex items-center justify-center rounded-radius-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter careers-section__cta-button"
                data-testid="cta-link"
                href="https://bluedot.org/ai-safety-course-operations/"
              >
                <span
                  class="cta-button__text"
                >
                  Apply now
                </span>
                <span
                  class="cta-button__chevron ml-3"
                >
                  <img
                    alt="→"
                    class="cta-button__chevron-icon size-2"
                    src="/icons/chevron_blue.svg"
                  />
                </span>
              </a>
            </div>
          </div>
          <div
            class="careers-section__listing"
          >
            <div
              class="careers-section__card--desktop w-full flex flex-row items-center justify-between p-8 container-lined"
            >
              <strong
                class="careers-section__title basis-[33%]"
              >
                Software Engineering Contractor
              </strong>
              <p
                class="careers-section__location"
              >
                London, Remote
              </p>
              <p
                class="careers-section__type"
              >
                Permanent
              </p>
              <a
                class="cta-button flex items-center justify-center rounded-radius-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter careers-section__cta-button"
                data-testid="cta-link"
                href="https://bluedot.org/swe-contractor/"
              >
                <span
                  class="cta-button__text"
                >
                  Apply now
                </span>
                <span
                  class="cta-button__chevron ml-3"
                >
                  <img
                    alt="→"
                    class="cta-button__chevron-icon size-2"
                    src="/icons/chevron_blue.svg"
                  />
                </span>
              </a>
            </div>
          </div>
          <div
            class="careers-section__listing"
          >
            <div
              class="careers-section__card--desktop w-full flex flex-row items-center justify-between p-8 container-lined"
            >
              <strong
                class="careers-section__title basis-[33%]"
              >
                Economics of Transformative AI Teaching Fellow
              </strong>
              <p
                class="careers-section__location"
              >
                London, Remote
              </p>
              <p
                class="careers-section__type"
              >
                Permanent
              </p>
              <a
                class="cta-button flex items-center justify-center rounded-radius-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter careers-section__cta-button"
                data-testid="cta-link"
                href="https://bluedot.org/economics-of-tai-teaching-fellow/"
              >
                <span
                  class="cta-button__text"
                >
                  Apply now
                </span>
                <span
                  class="cta-button__chevron ml-3"
                >
                  <img
                    alt="→"
                    class="cta-button__chevron-icon size-2"
                    src="/icons/chevron_blue.svg"
                  />
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
